README.md

```diff
@@ -8,7 +8,121 @@ the latest version of the library via the CI system. To interactively generate
+  Block-Jacobi preconditioner. This produces a preconditioner-specific type of
+  data.
++ conversions: benchmark conversions between matrix formats. This produces a
+  SuiteSparse type of benchmark data.
++ blas: benchmark Ginkgo dense BLAS functionality, like dot products, etc. This
+  produces an array of data points for different synthetic sizes.
++ sparse_blas: a benchmark of Ginkgo Sparse BLAS functionality, like SpGEMM. 
+
+
+### SuiteSparse data format
+Since it is the most common data type, we mostly describe the SuiteSparse type
+of benchmark data. The other benchmark data types are usually similar but
+simpler.
+
+For SuiteSparse data type, every matrix is in a different `.json` file. They can
+easily be put together into a large array of data points. For each matrix, the
+following data are always available:
++ filename: the full path to the matrix file that was benchmarked
++ problem: information about the matrix itself, like its unique SuiteSparse
+  `id`, the `name` of the matrix, the `group` it is part of, etc, but also simple
+  statistics about the row and column distribution of the nonzero elements in
+  `row_distribution` and `col_distribution`.
+
+The following data are benchmark dependent:
++ spmv: contains a list of data named after the benchmarked SpMV format. The
+  memory consumption is available in `storage`, `completed` is true if the SpMV
+  format could be run successfully (e.g., did not run out of memory), and `time`
+  contains the time for each `repetition`.
+  + the `optimal` SpMV format is also set as the fastest SpMV format
++ conversions: contains a list of data points each name in the form
+  `source-destination` matrix formats. It contains `completed`, `repetitions`
+  and `time` similarly to the SpMV benchmark.
++ solver: each solver data is provided under its solver name. If the solver is
+  preconditioned, the preconditioner will be listed in the name. 
+  + `recurrent`, `true` and `implicit` residual norms can be provided if a
+    detailed benchmark was run (more time-consuming).
+  + Also for a detailed run, `iteration_timestamps` are also listed in a corresponding array.
+  + The two main subparts of the solver data are the `generate` which lists the
+    amount of time taken to generate the solver from its factory, and the
+    `apply` which tries to solve the problem with a specific Right Hand Side
+    (RHS). The norm of the RHS is given in `rhs_norm`.
+  + In `apply`, the number of `iterations` taken for solving and the `time` are
+    always provided.
+  + For both `generate` and `apply` in the case of a detailed run, the time
+    taken for every solver sub-component (kernel, copies, etc) is given under
+    `components`.
+  + The `repetitions` and `completed` are the same as for the SpMV benchmark.
+
+## Licensing
+
+The Ginkgo benchmark data is available under the [CC-BY license](LICENSE.md).
+All contributions to the project are added under this license.
```

Comment:
Maybe also note that by pushing/committing to this repository, you are agreeing to providing the data in CC-BY license.